File: UniEatsBackEnd/Models/FoodItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace UniEatsBackEnd.Models
{
    public class FoodItem
    {
        [Key]
        public int ItemId { get; set; }

        [Required, MaxLength(100)]
        public string Name { get; set; }

        [Required]
        [RegularExpression("(Drinks|Snacks|Meals)", ErrorMessage = "Invalid category.")]
        public string Category { get; set; }

        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "Price must be greater than 0.")]
        public decimal Price { get; set; }

        public string Description { get; set; }

        [Url]
        public string ImageUrl { get; set; }

        public bool Availability { get; set; } = true;

        [Range(0, int.MaxValue, ErrorMessage = "Stock quantity must be non-negative.")]
        public int StockQuantity { get; set; }
    }
}

File: UniEatsFrontEnd/UniEatsFrontEnd/src/models/food-item.ts
export interface FoodItem {
  itemId: number; // Unique identifier for the food item
  name: string; // Name of the food item, required, max length 100
  category: 'Drinks' | 'Snacks' | 'Meals' | 'Appetizers' | 'Desserts'; // Restricted to specific values
  price: number; // Price of the item, must be greater than 0
  description?: string; // Optional description
  imageUrl?: string; // Optional URL for the item's image, must be valid
  availability?: boolean; // Availability status, defaults to true
  stockQuantity: number; // Stock quantity, must be non-negative
  discount?: number; // Discount percentage, between 0 and 100, defaults to 0
}
